.github/actions/setup-ipfs/action.yml

```diff
@@ -0,0 +1,41 @@
+name: Setup IPFS
+description: Setup IPFS
+runs:
+  using: "composite"
+  steps:
+    - uses: ipfs/download-ipfs-distribution-action@v1
+      with:
+        name: go-ipfs
+        version: "${{ env.GO_IPFS_VER }}"
+    - uses: ipfs/download-ipfs-distribution-action@v1
+      with:
+        name: ipfs-cluster-ctl
+        version: "${{ env.CLUSTER_CTL_VER }}"
+    - name: Init IPFS daemon
+      run: |
+        # fix resolv - DNS provided by Github is unreliable for DNSLik/dnsaddr
+        sudo sed -i -e 's/nameserver 127.0.0.*/nameserver 1.1.1.1/g' /etc/resolv.conf
+        ipfs init --profile flatfs,server,randomports,lowpower
+        # make flatfs async for faster ci
+        ipfs config --json 'Datastore.Spec.mounts' "$(ipfs config 'Datastore.Spec.mounts' | jq -c '.[0].child.sync=false')"
```

Comment:
We can't init on ipfs daemon start with `--init` flag because we want to modify the config between init and start.